pkg/storage/utils/decomposedfs/node/locks.go

```diff
@@ -121,7 +122,17 @@ func (n Node) ReadLock(ctx context.Context) (*provider.Lock, error) {
 		appctx.GetLogger(ctx).Error().Err(err).Msg("Decomposedfs: could not decode lock file, ignoring")
 		return nil, errors.Wrap(err, "Decomposedfs: could not read lock file")
 	}
-	return lock, err
+
+	// lock already expired
+	if lock.Expiration != nil && time.Now().After(time.Unix(int64(lock.Expiration.Seconds), int64(lock.Expiration.Nanos))) {
+		if err = os.Remove(f.Name()); err != nil {
+			return nil, errors.Wrap(err, "Decomposedfs: could not remove expired lock file")
```

Comment:
I think the `return` is not correct here. It is not nice if the file lock can not be removed physically but in the context of the application, that is no error condition in regards of the lock from the callers perspective. 

So imho this should only place a warning in the logfile, and fall through to the _no lock found_ reply below.